framework/wazuh/core/tests/test_stats.py

```diff
@@ -4,21 +4,125 @@
         sys.modules['wazuh.rbac.orm'] = MagicMock()
         import wazuh.rbac.decorators
-
-        del sys.modules['wazuh.rbac.orm']
+        from wazuh.core import common, stats
+        from wazuh.core.exception import (WazuhError, WazuhException,
+                                          WazuhInternalError)
         from wazuh.tests.util import RBAC_bypasser
 
+        del sys.modules['wazuh.rbac.orm']
         wazuh.rbac.decorators.expose_resources = RBAC_bypasser
-        from wazuh.core.exception import WazuhError, WazuhInternalError
-        from wazuh.core import stats, common
+
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data', 'stats')
+
+
+@pytest.mark.parametrize('date_, data_list', [
+    (date(2019, 8, 13), ['15-571-3-2', '15--107--1483--1257--0']),
+    (date(2019, 8, 13), ['15-571-3-2']),
+    (date(2019, 8, 13), ['15--107--1483--1257--0']),
+    (date(2019, 8, 13), ['15'])
+])
+@patch('wazuh.core.stats.common.stats_path', new=test_data_path)
+def test_totals_(date_, data_list):
+    """Verify totals_() function works as expected"""
+    with patch('wazuh.core.stats.open', return_value=data_list):
+        affected = stats.totals_(date_)[1]
+        if affected:
+            for line in data_list:
+                data = line.split('-')
+                print(data)
+                if len(data) == 4:
+                    assert int(data[1]) == affected[0]['alerts'][0]['sigid']
+                    assert int(data[2]) == affected[0]['alerts'][0]['level'], 'Data do not match'
+                    assert int(data[3]) == affected[0]['alerts'][0]['times'], 'Data do not match'
+                else:
+                    data = line.split('--')
+                    if len(data) == 5:
+                        assert int(data[0]) == affected[0]['hour'], 'Data do not match'
+                        assert int(data[1]) == affected[0]['totalAlerts'], 'Data do not match'
+                        assert int(data[2]) == affected[0]['events'], 'Data do not match'
+                        assert int(data[3]) == affected[0]['syscheck'], 'Data do not match'
+                        assert int(data[4]) == affected[0]['firewall'], 'Data do not match'
+
+
+def test_totals_ko_():
+    """Verify totals_() function exception with data problems works"""
+    with patch('wazuh.core.stats.open', side_effect=IOError):
+        with pytest.raises(WazuhException, match=".* 1308 .*"):
+            stats.totals_(date(1996, 8, 13))
+
+    with patch('wazuh.core.stats.open', return_value=['15-571-3-2', '15--107--1483']):
+        result = stats.totals_(date(1996, 8, 13))
+        assert result[0]
+
+
+@patch('wazuh.core.common.stats_path', new=test_data_path)
+def test_weekly_():
+    """Verify weekly_() function works as expected"""
+    result = stats.weekly_()
+    assert 0 == result[0]['Sun']['interactions'], 'Data do not match'
+    for day in "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat":
```

Comment:
You could import `DAYS` from `core.stats` instead of hard-coding the days like that. In this case, week days will not change but it could be a problem with other type of data.

Reply:
I think this should stay that way in case someone change `DAYS` array from `core.stats` by mistake since we should test if this array is maintaining that format or not.